### PR TITLE
feat: add bootstrap gitea command

### PR DESCRIPTION
**Pull request by Azhovan**
This PR implement `bootstrap gitea` command feature requested here: https://github.com/fluxcd/flux2/issues/4080

output: 
```
 ./bin/flux bootstrap gitea --owner Azhovan --repository test3       
► connecting to gitea.com
► cloning branch "main" from Git repository "https://gitea.com/Azhovan/test3.git"
✔ cloned repository
► generating component manifests
✔ generated component manifests
✔ committed sync manifests to "main" ("a5860ca2f6919d98874461d97e6495f03e22b67e")
► pushing component manifests to "https://gitea.com/Azhovan/test3.git"
► installing components in "flux-system" namespace
✔ installed components
✔ reconciled components
► determining if source secret "flux-system/flux-system" exists
► generating source secret
✔ public key: ecdsa-sha2-nistp384 AAAAE2VjZHNhLXNoYTItbmlzdHAzzDQAAAAIbmlzdHAzODQAAABhBEN2Y/IZhv0aDh1zy1lhdlGPUjE01Eg5WgMtPwMFn9womCxMTR4tZDHgevQ0/xhrNblr4rUVPlqLEaKmbJX28sbOeKb4UKm5X2LDYv2YFTg1X1tXK0TOs48X4onWgdZyug==
✔ configured deploy key "flux-system-main-flux-system" for "https://gitea.com/Azhovan/test3"
► applying source secret "flux-system/flux-system"
✔ reconciled source secret
► generating sync manifests
✔ generated sync manifests
✔ committed sync manifests to "main" ("b2616e147e2fece39fba2e7eef7ccafe134be09e")
► pushing sync manifests to "https://gitea.com/Azhovan/test3.git"
► applying sync manifests
✔ reconciled sync configuration
◎ waiting for Kustomization "flux-system/flux-system" to be reconciled
✔ Kustomization reconciled successfully
► confirming components are healthy
✔ helm-controller: deployment ready
✔ kustomize-controller: deployment ready
✔ notification-controller: deployment ready
✔ source-controller: deployment ready
✔ all components are healthy

```

### CODE REVIEW

**Comment by makkes** [comment]
I haven't looked at the PR in detail, yet but it generally looks good. What we need in addition is a PR at fluxcd/website to add a Gitea-specific page under https://fluxcd.io/flux/installation/bootstrap/.

**Comment by Azhovan** [comment]
> I haven't looked at the PR in detail, yet but it generally looks good. What we need in addition is a PR at fluxcd/website to add a Gitea-specific page under https://fluxcd.io/flux/installation/bootstrap/.

sure, I will create a PR there as well.

**Comment by Azhovan** [comment]
> I haven't looked at the PR in detail, yet but it generally looks good. What we need in addition is a PR at fluxcd/website to add a Gitea-specific page under https://fluxcd.io/flux/installation/bootstrap/.

Added here: https://github.com/fluxcd/website/pull/1676

File: pkg/bootstrap/provider/factory.go
Comment (Azhovan) [reply pkg/bootstrap/provider/factory.go:55]: fixed.

**Comment by Azhovan** [comment]
@makkes 
is there any thing that need to be done here?

**Comment by makkes** [comment]
Sorry for the delay @Azhovan. @somtochiama and I tested it and it works as expected. I'll see why the CI check is failing, probably just re-run it and merge the PR as soon as everything's green. Then we can go over the website PR.

**Comment by makkes** [comment]
@Azhovan please don't merge `main` into this branch but rebase instead.

**Comment by Azhovan** [comment]
> @Azhovan please don't merge `main` into this branch but rebase instead.

I just rebased, can we run the checks again to see if it is failing or fixed?

**Comment by Azhovan** [comment]
actually I think run e2e pipeline also in my fork

**Comment by Azhovan** [comment]
From what I think the e2e is failing only because go mod files are not updated, 
I will update them

**Comment by Azhovan** [comment]
nevertheless it wont fail on my fork https://github.com/Azhovan/flux2/actions/runs/6720222904/job/18263324685

**Comment by makkes** [comment]
Please squash your commits before we can merge.

**Comment by Azhovan** [comment]
> Please squash your commits before we can merge.

Thanks, sure, 

e2e looked green by the way!

**Review by makkes — APPROVED** [review]

**Pull request merged**
